webapp/components/integrations/components/add_oauth_app.jsx

```diff
@@ -144,8 +144,8 @@ export default class AddOAuthApp extends React.Component {
 
         OAuthActions.registerOAuthApp(
             app,
-            () => {
-                browserHistory.push('/' + this.props.team.name + '/integrations/oauth2-apps');
+            (data) => {
+                browserHistory.push('/' + this.props.team.name + '/integrations/confirm?type=oauth2-apps&token=' + data.id + '&secret=' + data.client_secret + '&callback_urls=' + data.callback_urls.join(', '));
```

Comment:
Passing around secrets in the URL? Probably should figure out a better way to do this.